File: array_problems/addTwoNumbers/addTwoNumbers.py
class ArrayProblems:
    def addTwoNumbers(self, array1, array2):
        num1 = int(''.join(map(str, array1)))
        num2 = int(''.join(map(str, array2)))
        items = str(num1 + num2)[::-1]
        return [int(digit) for digit in items]    
        
        
    def addTwoNumbersOptimal(self, l1, l2):
        dummyHead = ListNode(0)
        curr = dummyHead
        carry = 0
        while l1 != None or l2 != None or carry != 0:
            l1Val = l1.val if l1 else 0
            l2Val = l2.val if l2 else 0
            columnSum = l1Val + l2Val + carry
            carry = columnSum // 10
            newNode = ListNode(columnSum % 10)
            curr.next = newNode
            curr = newNode
            l1 = l1.next if l1 else None
            l2 = l2.next if l2 else None
        return dummyHead.next
        


array_problems = ArrayProblems()
arr1 = [9, 9, 9, 9, 9, 9, 9]
arr2 = [9, 9, 9, 9]
addedNumbers = array_problems.addTwoNumbers(arr1, arr2)
print(addedNumbers)
addedNumbersOptimal = array_problems.addTwoNumbersOptimal(arr1, arr2)
print(addedNumbersOptimal)

File: array_problems/addTwoNumbers/addTwoNumbers.php
<?php
class ArrayProblems {
    
    public function addTwoNumbers($arr1, $arr2) {
        $array = explode(",", (strrev(intval(implode($arr1)) + intval(implode($arr2)))));
        return $array;
    }
}
$arrayProblems = new ArrayProblems();
$arr1 = [ 9,9,9,9,9,9,9 ];
$arr2 = [ 9,9,9,9 ];
$addedNumbers = $arrayProblems->addTwoNumbers($arr1, $arr2);
print_r($addedNumbers);